### PR TITLE
[Changelog] Add a note about conformance synthesis in same-file extensions.

### PR DESCRIPTION
**Pull request by huonw**

### CODE REVIEW

**Comment by huonw** [comment]
@swift-ci please smoke test

**Review by jrose-apple — APPROVED** [review]
LGTM

File: CHANGELOG.md [hunk 1]
```diff
@@ -57,6 +57,40 @@ Swift 5.0
...
+  struct Generic<Param> {
+      var property: Param
+  }
+  extension Generic: Equatable where Param: Equatable {}
+  ```
+
+  Code that wants to be as precise as possible should generally not
+  conditionally conform to `Codable` directly, but rather its two constituent
+  protocols, or else one can only (for instance) decode a `Generic<Param>` if
+  `Param` is `Encodable` in additional to the `Decodable`:
```
Comment (rudkx) [inline CHANGELOG.md]: additional?

File: CHANGELOG.md [hunk 2]
```diff
@@ -57,6 +57,40 @@ Swift 5.0
 Swift 4.2
 ---------
 
+* [SE-0185][]
+
+  Protocol conformances can now be synthesized in extensions in the same file as
```
Comment (rudkx) [inline CHANGELOG.md]: "can now be" -> "are now"

?

File: CHANGELOG.md [hunk 3]
```diff
@@ -57,6 +57,40 @@ Swift 5.0
...
+  the type definition, allowing automatic synthesis of conditional conformances
+  to `Hashable`, `Equatable` and `Codable` (both `Encodable` and
+  `Decodable`). For instance,
+
+  ```swift
+  struct Generic<Param> {
+      var property: Param
+  }
+  extension Generic: Equatable where Param: Equatable {}
```
Comment (rudkx) [inline CHANGELOG.md]: If this last part is the auto-synthesized part, it's not clear. You could put it into a comment with something like:
// Automatically synthesized:
//   extension Generic...

File: CHANGELOG.md [hunk 4]
```diff
@@ -57,6 +57,40 @@ Swift 5.0
...
+  ```
+
+  Code that wants to be as precise as possible should generally not
+  conditionally conform to `Codable` directly, but rather its two constituent
+  protocols, or else one can only (for instance) decode a `Generic<Param>` if
```
Comment (rudkx) [inline CHANGELOG.md]: "two constituent protocols" -> "two constituent protocols Foo and Bar" 

where Foo and Bar are the constituent protocols...

File: CHANGELOG.md (hunk 2 above)
Comment (huonw) [reply CHANGELOG.md]: I think "can" is right here: the synthesis doesn't always happen.

File: CHANGELOG.md (hunk 3 above)
Comment (huonw) [reply CHANGELOG.md]: Added:

    // Automatically synthesized inside the extension:
    // static func ==(lhs: Generic, rhs: Generic) -> Bool {
    //   return lhs.property == rhs.property
    // }

**Comment by huonw** [comment]
@swift-ci please smoke test

**Pull request merged**